src/webgpu/api/operation/command_buffer/render/culling_tests.spec.ts

```diff
@@ -0,0 +1,160 @@
+export const description = 'Test culling and rasterizaion state.';
+
+import { makeTestGroup } from '../../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../../gpu_test.js';
+
+class CullingTest extends GPUTest {
+  createPipelineForTest(
+    frontFace: GPUFrontFace,
+    cullMode: GPUCullMode,
+    format: GPUTextureFormat
+  ): GPURenderPipeline {
+    // Draw two triangles with different winding orders:
+    // 1. The top-left one is counterclockwise (CCW)
+    // 2. The bottom-right one is clockwise (CW)
+    const vertexModule = this.makeShaderModule('vertex', {
+      glsl: `#version 450
+            const vec2 pos[6] = vec2[6](vec2(-1.0f,  1.0f),
+                                        vec2(-1.0f,  0.0f),
+                                        vec2( 0.0f,  1.0f),
+                                        vec2( 0.0f, -1.0f),
+                                        vec2( 1.0f,  0.0f),
+                                        vec2( 1.0f, -1.0f));
+            void main() {
+                gl_Position = vec4(pos[gl_VertexIndex], 0.0, 1.0);
+            }`,
+    });
+
+    // gl_FragCoord of pixel(x, y) in framebuffer coordinate is (x + 0.5, y + 0.5). And we use
+    // RGBA8 format for the back buffer. So (gl_FragCoord.xy - vec2(0.5)) / 255 in shader code
+    // will make the pixel's R and G channels exactly equal to the pixel's x and y coordinates.
+    const fragmentModule = this.makeShaderModule('fragment', {
+      glsl: `#version 450
+      layout(location = 0) out vec4 fragColor;
+      void main() {
+         fragColor = vec4((gl_FragCoord.xy - vec2(0.5)) / 255, 0.0, 1.0);
+      }`,
+    });
+
+    return this.device.createRenderPipeline({
+      vertexStage: { module: vertexModule, entryPoint: 'main' },
+      fragmentStage: { module: fragmentModule, entryPoint: 'main' },
+      primitiveTopology: 'triangle-list',
+      rasterizationState: { frontFace, cullMode },
+      colorStates: [
+        {
+          format,
+          alphaBlend: { dstFactor: 'zero', srcFactor: 'one', operation: 'add' },
+          colorBlend: { dstFactor: 'zero', srcFactor: 'one', operation: 'add' },
+          writeMask: GPUColorWrite.ALL,
+        },
+      ],
+      sampleCount: 1,
```

Comment:
These are all defaults, don't need to specify them
```suggestion
      colorStates: [ { format } ],
```